src/formats/internal/reader.c

```diff
@@ -170,18 +195,27 @@ int8_t sd_peek_byte(sd_reader *reader) {
 int16_t sd_peek_word(sd_reader *reader) {
     int16_t d = 0;
     sd_peek_buf(reader, (char *)&d, 2);
+#ifdef BIG_ENDIAN_BUILD
+    d = __builtin_bswap16(d);
+#endif
     return d;
 }
 
 int32_t sd_peek_dword(sd_reader *reader) {
     int32_t d = 0;
     sd_peek_buf(reader, (char *)&d, 4);
+#ifdef BIG_ENDIAN_BUILD
+    d = __builtin_bswap32(d);
+#endif
     return d;
 }
 
 float sd_peek_float(sd_reader *reader) {
     float f = 0;
     sd_peek_buf(reader, (char *)&f, 4);
+#ifdef BIG_ENDIAN_BUILD
+    f = __builtin_bswap16(f);
+#endif
```

Comment:
[__builtin_bswap16](https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html) does not take an argument of type float, it takes a uint16_t.
The `__builtin_bswap16(f)` call is equivalent to `__builtin_bswap16((uint16_t)f)`, which is why the N64 build was encountering floating point errors.

Also: bswap16 is the wrong number of bits, should be bswap32.

(same as other review comment: please implement sd_peek_float in terms of sd_peek_dword)

Reply:
Well, the exceptions were definitely from doing float f = __buildin_bswap32((float)x);
The only reason this function is wrong is because nothing in the engine calls it, and I just didn't hit the issue.